gql_error_link/lib/gql_error_link.dart

```diff
@@ -0,0 +1,79 @@
+/// GQL Link to handle execution errors and exceptions
+library gql_error_link;
+
+import "dart:async";
+import "package:async/async.dart";
+import "package:gql_link/gql_link.dart";
+import "package:gql_exec/gql_exec.dart";
+
+/// A handler of GraphQL errors.
+typedef ErrorHandler = Stream<Response> Function(
+  Request request,
+  NextLink forward,
+  Response response,
+);
+
+/// A handler of Link Exceptions.
+typedef ExceptionHandler = Stream<Response> Function(
+  Request request,
+  NextLink forward,
+  LinkException exception,
+);
+
+/// [ErrorLink] allows interception of GraphQL errors (using [onError])
+/// and [LinkException]s (using [onException]).
+///
+/// In both cases [ErrorLink] transfers control over to the handler which may
+/// return a new stream to discard the original stream. If the handler returns
+/// `null`, the original stream is left intact and will be allowed to continue
+/// streaming new events.
+class ErrorLink extends Link {
+  final ErrorHandler onError;
```

Comment:
Do we want to call this onGraphQLErrors to make it explicitly clear that this is not an error in the link?

Reply:
Yeah. Done.